src/main/java/org/cryptomator/common/vaults/VaultListManager.java

```diff
@@ -73,11 +74,27 @@ public Vault add(Path pathToVault) throws IOException {
 		return get(normalizedPathToVault) //
 				.orElseGet(() -> {
 					Vault newVault = create(newVaultSettings(normalizedPathToVault));
+					setVaultScheme(newVault);
 					vaultList.add(newVault);
 					return newVault;
 				});
 	}
 
+	private void setVaultScheme(Vault vault) {
+		try {
+			var keyLoader = vault.getVaultSettings().keyLoader;
+			if (Objects.isNull(keyLoader.get())) {
+				var vaultConfig = vault.getVaultConfigCache().get();
+				var keyIdScheme = vaultConfig.getKeyId().getScheme();
+				keyLoader.set(keyIdScheme);
+			}
+		} catch (NoSuchFileException e) {
+			LOG.error("NoSuchFileException", e);
+		} catch (IOException e) {
+			throw new RuntimeException("Unexpected Exception", e);
+		}
```

Comment:
This is bad error handling: In one case we are just logging the error while in the other we are throwing an unchecked exception with no clear sign, why we handle the cases differently. Also, throwing an unchecked exception is a bad idea, since it is nowhere clearly handled leading to an undefined state of the app.

I suggest to throw the IOException in any case to the leading method and change the vault state to ERROR or something similar, since the config is needed for unlock anyway.